lib/gulp/ts.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const path = require('path');
+const gulp = require('gulp');
+const plugins = require('gulp-load-plugins')();
+const templates = require('../templates/templates');
+
+module.exports = function (config, browsersync) {
+	const uglifyOptions = {
+		preserveComments: 'license'
+	};
+
+	const renameOptions = {
+		suffix: '.min'
+	};
+
+	// Avoid uglifying map files
+	const errorHandler = function (e) {
+		if (!path.extname(e.fileName) === '.map') {
+			console.error(e);
+		}
+	};
+
+	let pkg;
+	let banner;
+
+	if (config.header.enabled) {
+		pkg = require(path.resolve(config.header.packageJsonPath));
+		banner = templates.banner().join('\n');
+	}
+
+	// TypeScript
+	gulp.task('ts-gulp', () => {
+		return gulp.src(config.srcFolder + '/js/*.{ts,js}')
+			.pipe(plugins.plumber())
+			.pipe(plugins.if(config.header.enabled, plugins.header(banner, {pkg})))
+			.pipe(plugins.sourcemaps.init())
+			.pipe(plugins.typescript({
+				noImplicitAny: true,
+				allowJs: true
+			}))
+			.pipe(plugins.if(config.js.tsConcat, plugins.concat(config.js.tsConcatFilename + '.js')))
```

Comment:
I think the typescript plugin should not concatenate. I changed it to use the concat plugin for consistency.